### PR TITLE
Switch to Vite

### PR DESCRIPTION
**Pull request by GIT-Gizmo**
I have successfully migrated CRA to Vite. You can merge and check it out

### CODE REVIEW

**Comment by wweitzel** [comment]
Hey! Thanks, `npm run build` is not working on this PR. In `tsconfig.json` we need to set `"module": "ESNext"` and we also need to add `"moduleResolution": "node"`. Then it should work.

File: package.json [hunk 1]
```diff
@@ -15,15 +15,15 @@
...
     "typescript": "^4.8.4",
     "web-vitals": "^2.1.4"
   },
   "scripts": {
-    "start": "BROWSER=none react-scripts start",
-    "build": "react-scripts build",
-    "test": "react-scripts test",
-    "eject": "react-scripts eject",
+    "start": "vite",
+    "build": "tsc && vite build",
```
Comment (wweitzel) [inline package.json:23]: Let's change this line to `"build": "tsc --noEmit && vite build",` 

The `--noEmit` part makes it so js files are not added when running `npm run build`.

**Comment by wweitzel** [comment]
Two more things:

1. You will need to run `npm run format` and commit the changes to get this to pass CI
2. I'm not entirely sure what the new `vite-env.d.ts` file is for. But everything still seems to work if I remove it. Can you remove that file and commit the change? I'd prefer not to have it if we don't need it.

**Comment by GIT-Gizmo** [comment]
Sure, I'll do that 👍

**Comment by GIT-Gizmo** [comment]
Done

**Comment by wweitzel** [comment]
Thanks! I made some comments for changes that need to be made for this to work. After you address those, we should be good to go. Let me know if you have any questions. The comments are these

- https://github.com/wweitzel/top90-frontend/pull/39#issuecomment-1700052688
- https://github.com/wweitzel/top90-frontend/pull/39#discussion_r1310967621

**Comment by GIT-Gizmo** [comment]
Oh ok, I've done that. I believe it should be good now👍

**Comment by wweitzel** [comment]
Thanks! There is a small issue with the tests but I can fix that later. Im merging this now. Closes #28

**Review by wweitzel — APPROVED** [review]
LGTM

**Comment by GIT-Gizmo** [comment]
Awesome, glad I could help in my own little way

**Pull request merged**